R/align_runs.R

```diff
@@ -0,0 +1,257 @@
+                                     control=loess.control(surface="direct"))
+          loessFits[[pair]] <- Loess.fit
+        }
+        # Set up constraints for penalizing similarity matrix
+        rse <- min(Loess.fit$s, expRSE)
+        noBeef <- ceiling(RSEdistFactor*rse/samplingTime)
+        tVec.ref <- XICs.ref[[1]][["time"]] # Extracting time component
+        tVec.eXp <- XICs.eXp[[1]][["time"]] # Extracting time component
+        B1p <- predict(Loess.fit, tVec.ref[1])
+        B2p <- predict(Loess.fit, tVec.ref[length(tVec.ref)])
+        # Perform dynamic programming for chromatogram alignment
+        intensityList.ref <- lapply(XICs.ref, `[[`, 2) # Extracting intensity values
+        intensityList.eXp <- lapply(XICs.eXp, `[[`, 2) # Extracting intensity values
+        Alignobj <- alignChromatogramsCpp(intensityList.ref, intensityList.eXp,
+                                          alignType = alignType, tVec.ref, tVec.eXp,
+                                          normalization = normalization, simType = simMeasure,
+                                          B1p = B1p, B2p = B2p, noBeef = noBeef,
+                                          goFactor = goFactor, geFactor = geFactor,
+                                          cosAngleThresh = cosAngleThresh, OverlapAlignment = OverlapAlignment,
+                                          dotProdThresh = dotProdThresh, gapQuantile = gapQuantile,
+                                          hardConstrain = hardConstrain, samples4gradient = samples4gradient)
+        AlignedIndices <- cbind(Alignobj@indexA_aligned,
+                                Alignobj@indexB_aligned,
+                                Alignobj@score)
+        colnames(AlignedIndices) <- c("indexAligned.ref", "indexAligned.eXp", "score")
+        AlignedIndices[, 1:2][AlignedIndices[, 1:2] == 0] <- NA
+        tAligned.ref <- mapIdxToTime(tVec.ref, AlignedIndices[,"indexAligned.ref"])
+        tAligned.eXp <- mapIdxToTime(tVec.eXp, AlignedIndices[,"indexAligned.eXp"])
+        # Map retention time from reference to eXp.
+        rtTbl[pepIdx, eXp] <- tAligned.eXp[which.min(abs(tAligned.ref - rtTbl[pepIdx, ref]))]
+        df <- oswFiles[[eXp]] %>% dplyr::filter(transition_group_id == peptide) %>%
+          dplyr::select(leftWidth, rightWidth, RT, Intensity, peak_group_rank, m_score)
+        adaptiveRT <- RSEdistFactor*rse
+        df <- df %>% dplyr::filter(abs(RT - rtTbl[pepIdx, eXp]) <= adaptiveRT)
+        df <- df %>% dplyr::filter(m_score < maxFdrQuery & peak_group_rank == min(peak_group_rank)) %>% as.matrix()
+        if(nrow(df)==1){
+          # A feature is found. Use this feature for quantification.
+          lwTbl[pepIdx, eXp] <- df[1, "leftWidth"]
+          rtTbl[pepIdx, eXp] <- df[1, "RT"]
+          rwTbl[pepIdx, eXp] <- df[1, "rightWidth"]
+          intesityTbl[pepIdx, eXp] <- df[1, "Intensity"]
+        } else {
+          # Feature is not found.}
```

Comment:
@shubham1637 what to do i n this case? Integrate the raw chromatogram data?

Reply:
Yes. So far I only have integration of raw peak data. Will implement background subtraction and other methods to be consistent with OpenSWATH.